modules/azure/src/test/java/org/testcontainers/containers/CosmosDBEmulatorContainerTest.java

```diff
@@ -0,0 +1,55 @@
+package org.testcontainers.containers;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.models.CosmosContainerResponse;
+import com.azure.cosmos.models.CosmosDatabaseResponse;
+import org.assertj.core.api.Assertions;
+import org.junit.Rule;
+import org.junit.Test;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.FileOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+
+public class CosmosDBEmulatorContainerTest {
+
+    @Rule
+    // emulatorContainer {
+    public CosmosDBEmulatorContainer emulator = new CosmosDBEmulatorContainer(
+            DockerImageName.parse("mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator:2.11.13-c5c131e3")
+    );
+    // }
+
+    @Test
+    public void testWithCosmosClient() throws Exception {
+        // buildAndSaveNewKeyStore {
+        Path keyStoreFile = Files.createTempFile("azure-cosmos-emulator", ".keystore");
```

Comment:
Perhaps we should be using JUnit's [`TemporaryFolder`](https://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html)

Reply:
Ok, I'm looking into it and will rebase PR to get rid of merge commits

Reply:
Done.